### PR TITLE
feat: add visibility configuration

### PR DESCRIPTION
**Pull request by N3RG4L**
Added Visibility property, to add support for Private mode : https://aws.amazon.com/blogs/mobile/introducing-private-apis-on-aws-appsync/

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-visibility

### CODE REVIEW

**Review by bboure — CHANGES_REQUESTED** [review]
@N3RG4L  Thank you for opening this PR  🙏 🎉

before we merge, could you please
- Add proper TypeScript  types for the new property: (something like `'GLOBAL' | 'PRIVATE'`  instead of just `string`)
- Add [validation](https://github.com/sid88in/serverless-appsync-plugin/blob/master/src/validation.ts)
- Add a few unit tests (generation of Cfn and validation)

**Comment by N3RG4L** [comment]
@bboure  thank you for your fast answer, and for the guidance.
I just added the requirements you mentioned. I hope I missed nothing else

File: package.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
-  "name": "serverless-appsync-plugin",
-  "version": "0.0.0-development",
+  "name": "serverless-appsync-plugin-private",
+  "version": "1.0.0",
```
Comment (bboure) [inline package.json:3]: This change looks wrong and will break npm publish

File: src/validation.ts [hunk 2]
```diff
@@ -22,6 +22,11 @@ const DATASOURCE_TYPES = [
   'AMAZON_EVENTBRIDGE',
 ] as const;
 
+const VISIBILITY = [
+  'GLOBAL',
+  'PRIVATE',
+] as const;
```
Comment (bboure) [inline src/validation.ts:28]: is this used?

File: src/__tests__/api.test.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ describe('Api', () => {
                 },
               ],
               "XrayEnabled": false,
+              "Visibility": true,
```
Comment (bboure) [inline src/__tests__/api.test.ts:26]: Shouldn't this be `GLOBAL` or `undefined` ? 
Where does `true` come from?

File: src/validation.ts (hunk 2 above)
Comment (N3RG4L) [reply src/validation.ts]: no, forgot to delete. deleted

File: src/__tests__/api.test.ts (hunk 3 above)
Comment (N3RG4L) [reply src/__tests__/api.test.ts]: I replaced the test with "PRIVATE"

**Comment by N3RG4L** [comment]
added the requested changes

**Comment by bboure** [comment]
I added a few tests and made a few adjustments.
Thank you for your contribution 🎉

**Review by bboure — APPROVED** [review]

**Pull request merged**